doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,801 @@
+```
+;Stores STP interface details per Instance
+key                 = MSTP_INSTANCE_PORT:"Instance"instanceid:ifname    ; instanceid|ifname with prefix MSTP_INSTANCE_PORT
+port_num            = 1*3DIGIT                                          ; port number of bridge port
+path_cost           = 1*9DIGIT                                          ; port path cost (1 to 200000000)
+priority            = 1*3DIGIT                                          ; port priority (0 to 240, DEF:128)
+port_state          = "state"                                           ; STP state - disabled, block, listen, learn, forward
+port_role           = "role"                                            ; STP port role - root, designated, blocking, alternate, master
+desig_root    	    = 16HEXDIG                                          ; designated root
+desig_cost   	    = 1*9DIGIT                                          ; designated cost
+desig_bridge  	    = 16HEXDIG                                          ; designated bridge
+desig_port    	    = 1*3DIGIT                                          ; designated port
+fwd_transitions     = 1*5DIGIT                                          ; number of forward transitions
+bpdu_sent           = 1*10DIGIT                                         ; bpdu transmitted
+bpdu_received       = 1*10DIGIT                                         ; bpdu received
+tcn_sent            = 1*10DIGIT                                         ; tcn transmitted
+tcn_received        = 1*10DIGIT                                         ; tcn received
+root_guard_timer    = 1*3DIGIT                                          ; root guard current timer value
+```
+
+### Existing Tables
+Following already present APP_DB tables are also used for implementation of MSTP:
+
+#### STP_PORT_STATE_TABLE 
+The table holds the state of a port i.e forwarding, learning, blocking with respect to each instance.
+#### STP_VLAN_INSTANCE_TABLE 
+The table holds the VLAN to instance mapping.
+#### STP_FASTAGEING_FLUSH_TABLE 
+The table informs when the FDB flushing is required. This is done in case of topology change where the mac entries in FDB become inconsistent and there is a need to flush these entries.
+
+# SAI
+
+## Existing SAI Attributes
+Following table shows the existing SAI Attributes that will be used:
+
+|**Component**|**SAI Attribute**|
+| :- | :- |
+|STP Instance|SAI_STP_ATTR_VLAN_LIST
+||SAI_STP_ATTR_BRIDGE_ID|
+||SAI_STP_ATTR_PORT_LIST|
+|STP Port|SAI_STP_PORT_ATTR_STP |
+||SAI_STP_PORT_ATTR_BRIDGE_PORT|
+||SAI_STP_PORT_ATTR_STATE|
+|STP Port States|SAI_STP_PORT_STATE_LEARNING|
+||SAI_STP_PORT_STATE_FORWARDING|
+||SAI_STP_PORT_STATE_BLOCKING|
+|VLAN STP Instance|SAI_VLAN_ATTR_STP_INSTANCE|
+|Switch STP Attributes|SAI_SWITCH_ATTR_DEFAULT_STP_INST_ID|
+||SAI_SWITCH_ATTR_MAX_STP_INSTANCE|
+
+## New SAI Attributes
+MSTP design requires one new attribute `SAI_HOSTIF_TRAP_TYPE_MSTP` for control trap packets which will be defined in saihostif.h.
```

Comment:
Please explain why SAI\_HOSTIF\_TRAP\_TYPE\_STP is not sufficient. The DMAC and LLC fields are identical for STP and MSTP.  Creating separate trap types for STP and MSTP would required the hardware to parse/match the BPDU protocol identifier.